### PR TITLE
SIMD-0140: Migrate Config Program to Core BPF

### PR DESCRIPTION
**Pull request by buffalojoec**
Migrate the Config program to Core BPF.

### CODE REVIEW

**Comment by buffalojoec** [comment]
Side note @joncinque I think this SIMD can be a good template for other program migrations as well.

**Review by CantelopePeel — COMMENTED** [review]
Hate to stall this one but we may need some more language here. Good opportunity to specify the program. 

Will add more comments.

**Review by joncinque — COMMENTED** [review]
Looks good overall! I agree with @CantelopePeel on the need to specify the program in some way, even if it's an IDL or link. I'll defer to Firedancer on the format they find most useful

File: proposals/xxxx-migrate-config-to-core-bpf.md [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+The program will be reimplemented in order to be compiled to BPF and executed by
+the BPF loader.
+
+The reimplemented program's ABI will exactly match that of the original.
+
+The reimplemented program's functionality will exactly match that of the
+original, differing only in compute usage.
+
+The program will be migrated to Core BPF using the procedure outlined in
+[SIMD 0088](https://github.com/solana-foundation/solana-improvement-documents/pull/88).
```
[the remark this answers is not in the document]
Comment (buffalojoec) [reply proposals/xxxx-migrate-config-to-core-bpf.md:47]: Now that you mention it, I think I can actually just do `[SIMD 0088](./0088-enable-core-bpf-programs.md)`.

File: proposals/0140-migrate-config-to-core-bpf.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+original, differing only in compute usage. Instead it will have dynamic compute
+usage based on the VM's compute unit meter.
+
+The program will be migrated to Core BPF using the procedure outlined in
+[SIMD 0088](./0088-enable-core-bpf-programs.md).
+
+The program's upgrade authority will be a multi-sig authority with keyholders
+from Anza Technology and may expand to include contributors from other validator
+client teams.
+In the future, this authority could be replaced by validator governance.
```
[the remark this answers is not in the document]
Comment (buffalojoec) [reply proposals/0140-migrate-config-to-core-bpf.md:53]: @joncinque mentioned that, too. I'll update this bit.

**Comment by buffalojoec** [comment]
@joncinque @CantelopePeel How best to specify the program within the SIMD?

Here's the IDL: https://github.com/solana-program/config/blob/main/program/idl.json

Do we just link it? Maybe add a `program-specs` folder? Lmk any ideas.

**Review by lheeger-jump — APPROVED** [review]
The spirit of this document is good. I think it will make sense to spend more time looking at the form of the actual program out of band with this proposal. When I have had time I have been examining those new programs. Testing, fuzzing, and audit is going to be the meat of this work so this document is largely for alignment.

The IDL also looks fine and seems to describe most of what is necessary to get to the next phase of development.

**Comment by 0xSol** [comment]
Merging as this SIMD reached consensus.

**Pull request merged**